### PR TITLE
Random checkpoint warp on eco/death, add toggle-able collectable counters

### PR DESCRIPTION
**Pull request by dallmeyer**
Requested in #74 
The "deaths" that we cause during checkpoint warps are not counted as deaths that increment `*collectable-interval*`.

Moved around where `runMods` and disabled the `suspend` call, which seems to be stable for me.

Added toggles in menu for displaying collectable counts:
- total (fuel+buzzer+money only) 
- warp, counting towards the warp interval

image
image

### CODE REVIEW

File: Checkpoint randomizer/data/goal_src/jak1/engine/mods/checkpoint-randomizer.gc [hunk 1]
```diff
@@ -297,7 +302,7 @@
 
   ;; current-seed and backup-checkpoints-used are updated at this point, commit to file in case of crash/closed window
   (commit-to-file *randomizer-settings*)
-  (suspend)
+  ;;(suspend)
```
Comment (dallmeyer) [inline Checkpoint randomizer/data/goal_src/jak1/engine/mods/checkpoint-randomizer.gc:305]: @Zedb0T do you remember why we added this here, was it for a specific case? it was causing issues for warping on `ecovent`s, and everything I've tested so far seems to work without it

File: Checkpoint randomizer/data/goal_src/jak1/engine/game/main.gc [hunk 2]
```diff
@@ -116,8 +116,6 @@
   )
 
 (defun movie? ()
-(runMods)
```
Comment (dallmeyer) [inline Checkpoint randomizer/data/goal_src/jak1/engine/game/main.gc:119]: There were some cases where `movie?` was called 3+ times in the same frame and that was causing issues trying to draw the counters on screen - the place I moved it to seems to be the main display loop so should be called once per frame as we intended

File: Checkpoint randomizer/data/goal_src/jak1/engine/gfx/font.gc [hunk 3]
```diff
@@ -720,7 +720,7 @@
               (.lvf vf16 (&-> fw size-st1 quad))
               (.lvf vf17 (&-> fw size-st2 quad))
               (.lvf vf18 (&-> fw size-st3 quad))
-              (let ((has-flag-size24 (logand flags 32)))
+              (let ((has-flag-size24 (logand flags (font-flags large))))
```
Comment (dallmeyer) [inline Checkpoint randomizer/data/goal_src/jak1/engine/gfx/font.gc:723]: no real difference, just easier to read

File: Checkpoint randomizer/data/goal_src/jak1/engine/game/game-info-h.gc [hunk 4]
```diff
@@ -192,10 +192,14 @@
    (death-pos             vector-array                  :offset-assert 312)
    (dummy                 basic                         :offset-assert 316)
    (auto-save-count       int32                         :offset-assert 320)
+   (green-eco-total       float                         :offset-assert 324)
```
Comment (Zedb0T) [inline Checkpoint randomizer/data/goal_src/jak1/engine/game/game-info-h.gc:195]: I think modifications like this should be in randomizer-settings no?

File: Checkpoint randomizer/data/goal_src/jak1/engine/game/main.gc [hunk 5]
```diff
@@ -651,6 +649,8 @@
       ;; start immediately after all process updates finish.
       (profiler-instant-event "display-loop-top")
 
+      (runMods)
+
```
Comment (Zedb0T) [inline Checkpoint randomizer/data/goal_src/jak1/engine/game/main.gc:653]: Honestly might be worth putting a function named (runsOncePerFrame) here or something then we can add anything we want to run once per frame to it, mods or anything else really.

File: Checkpoint randomizer/data/goal_src/jak1/engine/ui/hud-classes.gc [hunk 6]
```diff
@@ -746,7 +746,31 @@
          (gp-0 (-> s5-0 base))
          )
     (let ((s4-0 draw-string-xy))
-      (format (clear *temp-string*) "~D" (-> obj value))
+      (cond
```
Comment (Zedb0T) [inline Checkpoint randomizer/data/goal_src/jak1/engine/ui/hud-classes.gc:749]: Think all this should be in a function in mods/library or mods/randomizer as well then a call to the function placed here if enabled

**Review by Zedb0T — APPROVED** [review]

File: Checkpoint randomizer/data/goal_src/jak1/engine/game/game-info-h.gc (hunk 4 above)
Comment (dallmeyer) [reply Checkpoint randomizer/data/goal_src/jak1/engine/game/game-info-h.gc:195]: This is tracking total green-eco collected throughout the game, similar to total-fuel/money/buzzer so it's not necessarily related to randomizer

File: Checkpoint randomizer/data/goal_src/jak1/engine/game/main.gc (hunk 5 above)
Comment (dallmeyer) [reply Checkpoint randomizer/data/goal_src/jak1/engine/game/main.gc:653]: Yeah, that's more or less how I think of (runMods)

File: Checkpoint randomizer/data/goal_src/jak1/engine/ui/hud-classes.gc (hunk 6 above)
Comment (dallmeyer) [reply Checkpoint randomizer/data/goal_src/jak1/engine/ui/hud-classes.gc:749]: By default it still has vanilla flavor so it's all opt-in, not sure which flag we would use to enable/disable? Again this one isn't really randomizer specific

**Pull request merged**